airflow/providers/openlineage/utils/sql.py

```diff
@@ -0,0 +1,194 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+from collections import defaultdict
+from contextlib import closing
+from typing import TYPE_CHECKING, Dict, Iterator, List
+
+from attrs import define, field
+
+from openlineage.client.facet import SchemaDatasetFacet, SchemaField
+from openlineage.client.run import Dataset
+
+if TYPE_CHECKING:
+    from airflow.hooks.base import BaseHook
+
+
+logger = logging.getLogger(__name__)
+
+_TABLE_SCHEMA = 0
+_TABLE_NAME = 1
+_COLUMN_NAME = 2
+_ORDINAL_POSITION = 3
+# Use 'udt_name' which is the underlying type of column
+# (ex: int4, timestamp, varchar, etc)
+_UDT_NAME = 4
+# Database is optional as 5th column
+_TABLE_DATABASE = 5
```

Comment:
It may be worth to group these constants in an `IntEnum`, since they are related:
```
from enum import IntEnum


class Index(IntEnum):
    SCHEMA = 0
    TABLE = 1
    COLUMN_NAME = 2
    ORDINAL_POSITION = 3
    UDT_NAME = 4
    DATABASE = 5
```